src/Nethermind/Nethermind.Core/Authentication/JwtAuthentication.cs

```diff
@@ -175,6 +194,43 @@ public bool Authenticate(string? token)
         return _timestamper.UnixTime.SecondsLong < exp;
     }
 
+    private static string GetDefaultFilePath()
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            string? xdgDataDir = Environment.GetEnvironmentVariable("XDG_DATA_HOME");
+
+            if (!string.IsNullOrEmpty(xdgDataDir))
+                return Path.Combine(xdgDataDir, "ethereum", "engine", "jwt.hex");
+
+            string? homeDir = Environment.GetEnvironmentVariable("HOME");
+
+            if (string.IsNullOrEmpty(homeDir))
+                throw new Exception("HOME environment variable is not set");
+
+            return Path.Combine(homeDir, ".local", "share", "ethereum", "engine", "jwt.hex");
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            string appDataDir = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+
+            return Path.Combine(appDataDir, "Ethereum", "Engine", "jwt.hex");
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            string? homeDir = Environment.GetEnvironmentVariable("HOME");
+
+            if (string.IsNullOrEmpty(homeDir))
+                throw new Exception("HOME environment variable is not set");
+
+            return Path.Combine(homeDir, "Library", "Application Support", "Ethereum", "Engine", "jwt.hex");
+        }
+
+        throw new Exception("Unsupported OS");
```

Comment:
Btw didn't supply why, this is a good read about it: https://stackoverflow.com/questions/22453650/why-are-we-not-to-throw-these-exceptions

Reply:
No, sorry. That's on me, been awhile since I did any c# code and i should have thrown a Todo Not implemented to showcase in the code review that is was a placeholder. But appreciate the resources.

Reply:
No worries, I just want for everyone to understand the why and grow and not just coming as picky :)